src/CondenserDotNet.Client/Configuration/ConsulProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+
+namespace CondenserDotNet.Client.Configuration
+{
+    public class ConsulProvider : ConfigurationProvider
+    {
+        private readonly IConfigurationRegistry _configurationRegistry;
+
+        public ConsulProvider(IConfigurationRegistry configurationRegistry)
+        {
+            _configurationRegistry = configurationRegistry;
+            _configurationRegistry.AddWatchOnEntireConfig(Load);
+        }
+
+        public override bool TryGet(string key, out string value)
+        {
+            return _configurationRegistry.TryGetValue(key, out value);
+        }
+
+        public override void Set(string key, string value)
+        {
+            _configurationRegistry.SetKeyAsync(key, value).Wait();
+        }
+
+        public override void Load()
+        {
+            OnReload();
+        }
+
+        public override IEnumerable<string> GetChildKeys(
+            IEnumerable<string> earlierKeys,
+            string parentPath)
+        {
+            var prefix = parentPath == null ? string.Empty : parentPath + ConfigurationPath.KeyDelimiter;
+
+            //Need to override this as we are not setting base Data, so expose all keys on registry
+            return _configurationRegistry.AllKeys
+                .Where(key => key.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
+                .Select(key => Segment(key, prefix.Length))
```

Comment:
Nested linq, I am not sure how often GetChildKeys is actually called, if it's not frequent it isn't worth optimizing right now